### PR TITLE
Enhancement: Redesigned search UI

### PR DESCRIPTION
**Pull request by sivin-git**
Redesigned the search UI to match the overall look and feel.

**After**

https://user-images.githubusercontent.com/64252451/111060998-f98b9e80-84c6-11eb-9327-d8df21ac797b.mov

**Before**

Screenshot 2021-03-02 at 10 38 49 AM

### CODE REVIEW

File: app/javascript/dashboard/routes/dashboard/conversation/search/PopOverSearch.vue [hunk 1]
```diff
@@ -0,0 +1,224 @@
+<template>
+  <div class="search-header">
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/conversation/search/PopOverSearch.vue:2]:
```suggestion
  <div class="search-wrap">
```

File: app/javascript/dashboard/routes/dashboard/conversation/search/PopOverSearch.vue [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    focus: {
+      inserted(el) {
+        el.focus();
+      },
+    },
+  },
+
+  mixins: [timeMixin, messageFormatterMixin, clickaway],
+
+  props: {},
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/conversation/search/PopOverSearch.vue:80]: Remove this line

File: app/javascript/dashboard/routes/dashboard/conversation/search/PopOverSearch.vue [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+              <span class="message-counter">({{ resultsCount }})</span>
+            </p>
+            <div v-if="uiFlags.isFetching" class="search--activity-message">
+              <woot-spinner size="" />
+              {{ $t('CONVERSATION.SEARCH.LOADING_MESSAGE') }}
+            </div>
+          </div>
+
+          <div
+            v-if="searchTerm && conversations.length && !uiFlags.isFetching"
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/conversation/search/PopOverSearch.vue:30]: Move this condition inside `v-if` as a new computed function

File: app/javascript/dashboard/routes/dashboard/conversation/search/PopOverSearch.vue [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+              :user-name="conversation.sender_name"
+              :timestamp="conversation.created_at"
+              :message="conversation.content"
+              :search-term="conversation.content"
+              :message-type="conversation.message_type"
+            />
+          </div>
+          <div
+            v-else-if="
+              searchTerm && !conversations.length && !uiFlags.isFetching
```
Comment (nithindavid) [inline app/javascript/dashboard/routes/dashboard/conversation/search/PopOverSearch.vue:46]: Reuse the above computed function created in previous comment

**Review by nithindavid — DISMISSED** [review]

File: app/javascript/dashboard/routes/dashboard/conversation/search/PopOverSearch.vue [hunk 5]
```diff
@@ -12,12 +12,14 @@
...
     </div>
     <div v-if="showSearchBox" class="results-wrap">
-      <div class="results">
+      <div class="show-results">
         <div>
           <div class="result-view">
             <p class="result">
               Search Results
-              <span class="message-counter">({{ resultsCount }})</span>
+              <span v-if="resultsCount !== 0" class="message-counter">
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/conversation/search/PopOverSearch.vue:20]: `!resultsCount` is enough

File: app/javascript/dashboard/routes/dashboard/conversation/search/SearchMessageItem.vue [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+  }
+}
+
+.search-message {
+  border-bottom: 1px solid #eee;
+}
+
+.user-wrap {
+  display: flex;
+  align-items: last baseline;
```
Comment (pranavrajs) [inline app/javascript/dashboard/routes/dashboard/conversation/search/SearchMessageItem.vue:106]: last?

File: app/javascript/dashboard/store/modules/conversationSearch.js [hunk 7]
```diff
@@ -11,6 +11,7 @@ export const getters = {
   getConversations(state) {
     return state.records;
   },
+
```
Comment (pranavrajs) [inline app/javascript/dashboard/store/modules/conversationSearch.js:14]: Unnecessary line.

**Review by pranavrajs — CHANGES_REQUESTED** [review]
Nice work on breaking the search component into smaller components. I have some minor comments, please see the individual review comments.

**Pull request merged**